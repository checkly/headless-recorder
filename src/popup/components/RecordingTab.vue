<template>
  <div class="tab recording-tab">
    <div class="content">
      <div class="empty" v-show="!isRecording">
        <img src="/images/PlayWright.png" alt="logo" width="78px">
        <h3>No recorded events yet</h3>
        <p class="text-muted">Click record to begin</p>
      </div>
      <div class="events" v-show="isRecording">
        <p class="text-muted text-center loading" v-show="liveEvents.length === 0">Waiting for events</p>
        <ul class="event-list">
          <li v-for="(event, index) in liveEvents" :key="index" class="event-list-item">
            <div class="event-label">
              {{index + 1}}.
            </div>
            <div class="event-description">
              <div class="event-action">{{event.action}}</div>
              <div class="event-props text-muted">{{event.selector || parseEventValue(event)}}</div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</template>
<script>
  export default {
    name: 'RecordingTab',
    props: {
      isRecording: { type: Boolean, default: false },
      liveEvents: { type: Array, default: () => { return [] } }
    },
    methods: {
      parseEventValue (event) {
        if (event.action === 'viewport*') return `width: ${event.value.width}, height: ${event.value.height}`
        if (event.action === 'goto*') return event.href
        if (event.action === 'navigation*') return ''
      }
    }
  }
</script>
<style lang="scss" scoped>
  @import "~styles/_animations.scss";
  @import "~styles/_variables.scss";


  .recording-tab {
    .content {
      display:flex;
      flex-direction:column;
      height:100%;
      min-height: 200px;
      .empty {
        padding: $spacer;
        text-align: center;
      }

      .events {
        max-height: $max-content-height;
        flex: 1;
        height:100%;
        overflow: auto;
        display: flex;
        flex-direction: column-reverse;

        .loading:after {
          content: '.';
          animation: dots 1s steps(5, end) infinite;
          animation-delay: 1.5s;
          margin-bottom: auto;
        }

        .event-list {
          list-style-type: none;
          padding: 0;
          margin: 0;

          .event-list-item {
            padding: 12px;
            font-size: 12px;
            border-top: 1px solid $gray-light;
            display: flex;
            flex: 1 1 auto;
            height: 32px;

            .event-label {
              vertical-align: top;
              margin-right: $spacer;
            }

            .event-description {
              margin-right: auto;
              display: inline-block;

              .event-action {
                font-weight: bold;
              }

              .event-props {
                white-space: pre;
              }
            }

          }
        }
      }
    }
    .nag-cta {
      a {
        color: $pink;
        font-size: 80%;
        font-weight: 500;
      }
    }
  }
</style>
