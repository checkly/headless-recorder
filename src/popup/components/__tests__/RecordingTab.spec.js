import { mount } from '@vue/test-utils'
import RecordingTab from '../RecordingTab.vue'

describe('RecordingTab.vue', () => {
  test('it has the correct pristine / empty state', () => {
    const wrapper = mount(RecordingTab)
    expect(wrapper.element).toMatchSnapshot()
  })

  test('it has the correct waiting for events state', () => {
    const wrapper = mount(RecordingTab)
    wrapper.setProps({ isRecording: true })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.event-list').isEmpty()).toBe(true)
  })

  test('it has the correct recording Puppeteer custom events state', () => {
    const wrapper = mount(RecordingTab)
    wrapper.setProps({
      isRecording: true,
      liveEvents: [{
        action: 'goto*',
        href: 'http://example.com'
      }, {
        action: 'viewport*',
        selector: undefined,
        value: { width: 1280, height: 800 }
      }, {
        action: 'navigation*',
        selector: undefined
      }]
    })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.event-list').isEmpty()).toBe(false)
  })

  test('it has the correct recording DOM events state', () => {
    const wrapper = mount(RecordingTab)
    wrapper.setProps({ isRecording: true, liveEvents: [{ action: 'click', selector: '.main > a.link', href: 'http://example.com' }] })
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.find('.event-list').isEmpty()).toBe(false)
  })
})
