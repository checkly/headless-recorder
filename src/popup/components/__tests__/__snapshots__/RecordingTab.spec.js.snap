// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordingTab.vue it has the correct pristine / empty state 1`] = `
<div
  class="tab recording-tab"
>
  <div
    class="content"
  >
    <div
      class="empty"
    >
      <img
        alt="desert"
        src="/images/Desert.svg"
        width="78px"
      />
       
      <h3>
        No recorded events yet
      </h3>
       
      <p
        class="text-muted"
      >
        Click record to begin
      </p>
       
      <div
        class="nag-cta"
      >
        <a
          href="https://checklyhq.com/headless-recorder"
          target="_blank"
        >
          Puppeteer Recorder is now 
          <strong>
            Headless Recorder
          </strong>
           and supports Playwright →
        </a>
      </div>
    </div>
     
    <div
      class="events"
      style="display: none;"
    >
      <p
        class="text-muted text-center loading"
      >
        Waiting for events
      </p>
       
      <ul
        class="event-list"
      />
    </div>
  </div>
</div>
`;

exports[`RecordingTab.vue it has the correct recording DOM events state 1`] = `
<div
  class="tab recording-tab"
>
  <div
    class="content"
  >
    <div
      class="empty"
    >
      <img
        alt="desert"
        src="/images/Desert.svg"
        width="78px"
      />
       
      <h3>
        No recorded events yet
      </h3>
       
      <p
        class="text-muted"
      >
        Click record to begin
      </p>
       
      <div
        class="nag-cta"
      >
        <a
          href="https://checklyhq.com/headless-recorder"
          target="_blank"
        >
          Puppeteer Recorder is now 
          <strong>
            Headless Recorder
          </strong>
           and supports Playwright →
        </a>
      </div>
    </div>
     
    <div
      class="events"
      style="display: none;"
    >
      <p
        class="text-muted text-center loading"
      >
        Waiting for events
      </p>
       
      <ul
        class="event-list"
      />
    </div>
  </div>
</div>
`;

exports[`RecordingTab.vue it has the correct recording Puppeteer custom events state 1`] = `
<div
  class="tab recording-tab"
>
  <div
    class="content"
  >
    <div
      class="empty"
    >
      <img
        alt="desert"
        src="/images/Desert.svg"
        width="78px"
      />
       
      <h3>
        No recorded events yet
      </h3>
       
      <p
        class="text-muted"
      >
        Click record to begin
      </p>
       
      <div
        class="nag-cta"
      >
        <a
          href="https://checklyhq.com/headless-recorder"
          target="_blank"
        >
          Puppeteer Recorder is now 
          <strong>
            Headless Recorder
          </strong>
           and supports Playwright →
        </a>
      </div>
    </div>
     
    <div
      class="events"
      style="display: none;"
    >
      <p
        class="text-muted text-center loading"
      >
        Waiting for events
      </p>
       
      <ul
        class="event-list"
      />
    </div>
  </div>
</div>
`;

exports[`RecordingTab.vue it has the correct waiting for events state 1`] = `
<div
  class="tab recording-tab"
>
  <div
    class="content"
  >
    <div
      class="empty"
    >
      <img
        alt="desert"
        src="/images/Desert.svg"
        width="78px"
      />
       
      <h3>
        No recorded events yet
      </h3>
       
      <p
        class="text-muted"
      >
        Click record to begin
      </p>
       
      <div
        class="nag-cta"
      >
        <a
          href="https://checklyhq.com/headless-recorder"
          target="_blank"
        >
          Puppeteer Recorder is now 
          <strong>
            Headless Recorder
          </strong>
           and supports Playwright →
        </a>
      </div>
    </div>
     
    <div
      class="events"
      style="display: none;"
    >
      <p
        class="text-muted text-center loading"
      >
        Waiting for events
      </p>
       
      <ul
        class="event-list"
      />
    </div>
  </div>
</div>
`;
